src/h2n_generate.erl

```diff
@@ -0,0 +1,241 @@
+%%% ---------------------------------------------------------------------------
+%%% @doc
+%%%
+%%%  This module actually generates the final file. It uses John
+%%%  Hughes' brilliant `prettypr` module to output nice, human
+%%%  readable Nix expressions
+%%%  ---------------------------------------------------------------------------
+-module(h2n_generate).
+
+%%
+%% API
+%%
+-export([nix_expression/1]).
+
+-include("hex2nix.hrl").
+%%
+%% Types
+%%
+
+%% We almost never do this, however, with `prettypr` it actually makes
+%% some modicum of sense.
+-import(prettypr, [above/2, beside/2, sep/1, par/1, break/1,
+                   par/2, text_par/1, empty/0]).
+
+%% ============================================================================
+%% Exported Functions
+%% ============================================================================
+-spec nix_expression([h2n_fetcher:dep_desc()]) -> string().
+nix_expression(Deps) ->
```

Comment:
Thats the topo sort in h2n_fetcher (using h2n_topo). If we just want to an alpha sort thats pretty trivial and reduces the code as well.